.github/dependabot.yml

```diff
@@ -5,6 +5,13 @@ updates:
     schedule:
       interval: "monthly"
     open-pull-requests-limit: 10
+  - package-ecosystem: "gradle"
+    directory: "/"
+    allow:
+      - dependency-name: "com.gradle*"
```

Comment:
why did we pick this approach instead of the following that we do in our own repos?
```
registries:
  - gradle-plugin-portal
```

Reply:
I actually copied that from our repo 😅
https://github.com/gradle/common-custom-user-data-gradle-plugin/blob/main/.github/dependabot.yml

Reply:
I restricted the dependency name to apply a defensive approach as they are having a per folder configuration and if not mistaken, `dependabot` is visiting recursively the folder structure subtree.

Reply:
I am not exactly following. How is the per folder configuration related to which repository or the dependancy name?
i.e., what would go wrong if we followed the approach used in our solutions repositories?

Reply:
Dependabot is analyzing dependencies on all the sub hierarchy, the one I'm adding is applied to /.
This basically means that having a fine grained per folder configuration is becoming useless if you don't filter the deps on `com.gradle*`

It would be simpler but I presume that they want to keep their current mechanism.

Reply:
Sorry, I still don't see the connection to the folders, but can you answer the following questions:

I think that we should only use this to provide updates to the Gradle Enterprise Gradle Plugin and the Common Custom User Data Gradle Plugin. Will this achieve that?
Why is the following line not required here?
```
registries:
  - gradle-plugin-portal
```

Reply:
My experience is that this is not required for public registries
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file?learn=dependency_version_updates#registries

The plugin portal is apparently hardcoded into dependabot-core
https://github.com/dependabot/dependabot-core/blob/20ed6c1166569c243cc55a55e09cf9109e31f1e8/gradle/lib/dependabot/gradle/file_parser/repositories_finder.rb#L16

This update configuration has been tested and allows to fetch new versions of GE Gradle plugin and CCUD.

Reply:
Agreed, let's be conservative here 👍

Reply:
Applying the registry requires defining it 😎
This adds an extra section which is not super nice (dummy login/pwd)
Let me know if you think we should keep it like this but I'm not quite sure.

Reply:
Ah, i see the complication now in adding the dummy login/pwd. Thanks for checking that. I think it makes sense to be conservative here.